src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1214,8 +1215,14 @@ func (k *kataAgent) appendVfioDevice(dev ContainerDevice, device api.Device, c *
 		kataDevice.Type = kataVfioGuestKernelDevType
 	}
 
-	for i, pciDev := range devList {
-		kataDevice.Options[i] = fmt.Sprintf("0000:%s=%s", pciDev.BDF, pciDev.GuestPciPath)
+	// The agent must handle VFIO-PCI and VFIO-AP devices differently.
+	// Prepend the options with `ap:` or `pci:` accordingly.
+	for i, device := range devList {
+		if device.IsAP {
+			kataDevice.Options[i] = fmt.Sprintf("ap:%s", strings.Join(device.APDevices, ","))
+		} else {
+			kataDevice.Options[i] = fmt.Sprintf("pci:0000:%s=%s", device.BDF, device.GuestPciPath)
```

Comment:
This suggests "AP" trumps "PCI" right?

Reply:
Because of the order? I had just used that order to have
```
if cond {
  do_cond_stuff()
} else {
  do_non_cond_stuff()
}
```

instead of
```
if !cond {
  do_non_cond_stuff()
} else {
  do_cond_stuff()
}
```

but this would change with the enum anyway

Reply:
As I noted in the agent code, I think using a different `kataDevice.Type` value would make more sense than putting this into all the options.  You're also making things unnecessarily different from the PCI case by folding all the AP devices into a single option, whereas PCI devices are placed one per option.